### DISCUSSION TITLE
Image delete on garbage collection

### DISCUSSION THREAD

**Discussion by jmauzyk** [verbb/formie#1258] [Q&A]
For forms set to delete uploads on submission deletion, if a submission is trashed and then restored, the images are no longer available. Wondering if there's a way to delete uploads when their submission is permanently deleted instead, or if it's just a technical limitation that they have to be deleted according to the current behavior.

**Comment by engram-design** [comment]
Yeah, that's tricky and not possible at the moment with Craft. There's no Craft event to know when something in permanently deleted, otherwise we would certainly trigger this action when things are **permanently** deleted, not soft-deleted. Unfortunately, we need to ensure assets are deleted, and the only way to do that is on soft-delete.

https://github.com/craftcms/cms/discussions/8745

**Comment by engram-design** [comment]
Updated in [4.0.0-beta.6](https://github.com/verbb/formie/releases/tag/4.0.0-beta.6).
